src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+                        pkgs += ['sssd-' + id_provider]
+                    end
+                }
+            end
+            if @ldap_pam
+                pkgs += ['pam_ldap']
+            end
+            if @krb_pam
+                pkgs += ['pam_krb5', 'krb5', 'krb5-client']
+            end
+            if @ldap_nss
+                pkgs += ['nss_ldap']
+                if @ldap_nss.include?('automount')
+                    pkgs += ['openldap2-client'] # provides /etc/openldap/ldap.conf
+                end
+            end
+            if @autofs_enabled || @sssd_nss.include?('automount') || @ldap_nss.include?('automount')
+                pkgs += ['autofs']
+            end
+            if @nscd_enabled
+                pkgs += ['nscd']
+            end
+            return pkgs
+        end
+
+        # Summarise the authentication configurations in a human-readable, single line of text.
+        def summary_text
+            # Figure out how authentication works on this computer
+            auth_doms_caption = ''
+            if !@sssd_enabled && @ldap_nss.empty? && !@ldap_pam && !@krb_pam
+                # Local only
+                auth_doms_caption = _('Only use local authentication')
+            elsif @sssd_enabled && (@sssd_pam || @sssd_nss.any?)
+                # SSSD is configured
+                if Service.Active('sssd')
+                    auth_doms_caption = @sssd_conf['sssd']['domains'].join(', ')
+                else
+                    auth_doms_caption = _('Configured but not activated')
+                end
+            else
+                # LDAP and/or Kerberos is configured
+                if @ldap_nss.any? || @ldap_pam
+                    if @ldap_conf['base'].to_s == ''
+                        auth_doms_caption = _('LDAP is enabled but the setup is incomplete')
+                    else
+                        auth_doms_caption = _('via LDAP on ') + @ldap_conf['base']
+                    end
+                end
+                if @krb_pam
+                    if auth_doms_caption != ''
+                        # 'and' as in "authenticate via LDAP and Kerberos"
+                        auth_doms_caption += _(' and ')
+                    end
+                    realms = @krb_conf.fetch('realms', {})
+                    if realms.length == 0
+                        auth_doms_caption += _('via Kerberos')
+                    else
+                        auth_doms_caption += _('via Kerberos on ') + realms.keys.join(', ')
+                    end
+                end
+            end
+            return auth_doms_caption
+        end
+
+        attr_accessor(:krb_conf, :krb_pam, :ldap_conf, :ldap_pam, :ldap_nss, :sssd_conf, :sssd_pam, :sssd_nss, :sssd_enabled)
+        attr_accessor(:autofs_enabled, :nscd_enabled, :mkhomedir_pam)
+        attr_accessor(:ad_domain, :ad_user, :ad_ou, :ad_pass, :ad_overwrite_smb_conf, :autoyast_editor_mode, :autoyast_modified)
```

Comment:
Attribute declarations should be at the top.
Also, there should be one per line with a YARD comment supplying the type and description of the attribute.
That way, the reader of the code will understand it much easier.